### PR TITLE
Piedram/appeals 37878

### PR DESCRIPTION
**Pull request by piedram**
Resolves:

# Description
As a Mail team user I need to fill out response letter information so that I can add the response letter to a correspondence. 
 
## Acceptance Criteria
- [ ] Code compiles correctly

When the "+add letters" button is clicked a response letter tile appears with the below sections:

1 Date Sent
  - This is defaulted to the current date
  - This has a date picker
  - Required
 
2 Letter Type: Dropdown
   - Is populated with values in attached document
   - Required

3 Letter Title 
   - This dropdown is disabled until a Letter Type is selected
   - Required

4 Letter sub-category 
  - This dropdown is disabled until a Letter Title is selected
  - Is populated with values in attached document
  - Required

5 Letter sub-category reason 
  - This dropdown is disabled until a sub-category is selected
  - Is populated with values in attached document
  - Required only if Letter sub-category has values

6 How long should the response window be for this response letter? 
   6.1  Radio button
   6.2 Custom value needs to be between 1-64
       - "Number of days (Value must be between 1 and 64)"
       - Text value
   6.3 Options are based on above selections in attached document
   6.4 Required

7 User can add additional response letters
  - User can add up to 3 response Letters

8 User can remove response letters

9 If a response letter button is clicked, "Continue" button is grayed out until all dropdowns are filled out or user removes response letter

## Testing Plan

1. Login as a Mail team user `MAIL_TEAM_SUPERVISOR_MAIL_INTAKE_USER (VACO)`
2. From the App selector, select and click on `Queue`
3. In Your cases page, click on `Switch views` and select `Correspondence Cases`
4. Now you have to see the `Correspondence Cases` page
5.  Click in one case, you can see the `Review Package` page
6. Scroll to the bottom of the page and click on `Create record` button
7. Now you can see the first tab on `intake` page and see the `+ Add Letter` button

image (28)

8. When you click on `+ Add Letter` a new letter appears, and the continue button disappears until you complete the entire form.
9. You can Add only 3 letters for correspondence or less when you have 3 letters the `+ Add Letter` button is not clickable anymore.
10. Use the `Remove letter` link to remove the letter.

image (29)

image (30)

11. The Windows response, depends on the `letter type` or `letter title` so when you click in this selector the response windows show up with the options

image (31)

image (32)

image (33)

image (34)

12. When the response window is `65 days` you have a custom option too.

image (35)

image (36)

### CODE REVIEW

File: client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+      disabled: stateOptions }
+  ];
+
+  const [valueOptions, setValueOptions] = useState(radioOptions);
+
+  const handleCustomWindowState = (currentOpt) => {
+    if (currentOpt === radioOptions[2].value) {
+      setResponseWindows(radioOptions[2].value);
+      setCustomResponseWindowState(true);
+      setValueOptions(valueOptions);
```
Comment (vinner57) [inline client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx:118]: I don't see any change in valueOptions state, then why set it?

File: client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx [hunk 2]
```diff
@@ -0,0 +1,355 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+// import { connect } from 'react-redux';
```
Comment (cacevesva) [inline client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx:3]: I think we should removed these comments out

File: client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+// import { bindActionCreators } from 'redux';
```
Comment (cacevesva) [inline client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx:4]: remove

File: client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+import TextField from '../../../../../components/TextField';
+import Button from '../../../../../components/Button';
+import SearchableDropdown from 'app/components/SearchableDropdown';
+import DateSelector from 'app/components/DateSelector';
+import RadioField from '../../../../../components/RadioField';
+import { ADD_CORRESPONDENCE_LETTER_SELECTIONS } from '../../../../constants';
+import moment from 'moment';
+
+// import {
```
Comment (cacevesva) [inline client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx:13]: remove

File: client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+      <div className="response_letter_date">
+        <DateSelector
+          name="date-set"
+          label="Date sent"
+          value= {date}
+          onChange={(val) => setDate(val)}
+          type="date"
+        />
+      </div>
+      <br />
```
Comment (vinner57) [inline client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx:274]: if these <br /> tags are being used just for vertical spacing, I would suggest using css padding, margin or some other properties.

File: client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+    for (let i = 0; i < valueOptions.length; i++) {
+      const option = valueOptions[i];
+
+      if (responseWindows === option.displayText) {
+        setStateOptions(false);
+        option.disabled = false;
+        if (responseWindows === '65 days') {
+          valueOptions[2].disabled = false;
+          valueOptions[1].disabled = true;
+          // setResponseWindows(radioOptions[0].value);
```
Comment (cacevesva) [inline client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx:182]: remove comment

**Review by vinner57 — CHANGES_REQUESTED** [review]

File: client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx (hunk 6 above)
Comment (piedram) [reply client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx:182]: done

File: client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx (hunk 4 above)
Comment (piedram) [reply client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx:13]: done

File: client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx (hunk 3 above)
Comment (piedram) [reply client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx:4]: done

File: client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx (hunk 2 above)
Comment (piedram) [reply client/app/queue/correspondence/intake/components/AddCorrespondence/AddLetter.jsx:3]: done

**Review by cacevesva — DISMISSED** [review]

**Pull request merged**